### ISSUE TITLE
Creating a lang file with the same name as a Nova resource causes exception

### ISSUE THREAD

**Issue by nathan-io** [laravel/nova-issues#2263]
- Laravel Version: 6.0.4
- Nova Version: 2.9.2
- PHP Version: 7.4.1
- Database Driver & Version: MySQL 8.0.18
- Operating System and Version: Ubuntu 18.04 LTS
- Browser type and version: n/a

### Description:

Example: We have a resource for our `Pages` model.

`app/Nova/Pages.php`:

```
<?php

namespace App\Nova;

use Illuminate\Http\Request;
use Laravel\Nova\Fields\ID;
use Laravel\Nova\Fields\BelongsTo;
use Laravel\Nova\Fields\Text;
use Laravel\Nova\Fields\Boolean;
use Laravel\Nova\Fields\DateTime;
use MrMonat\Translatable\Translatable;
use Inspheric\NovaDefaultable\HasDefaultableFields;

class Page extends Resource
{
    /**
     * The model the resource corresponds to.
     *
     * @var  string
     */
    public static $model = \App\Page::class;

    /**
     * The single value that should be used to represent the resource when being displayed.
     *
     * @var  string
     */
    public static $title = 'name';

    /**
     * The columns that should be searched.
     *
     * @var  array
     */
    public static $search = [
        'identifier', 'slug', 'name', 'id'
    ];

    /**
     * Get the displayable label of the resource.
     *
     * @return  string
     */
    public static function label()
    {
        return __('Pages');
    }

    /**
    * Get the displayable singular label of the resource.
    *
    * @return  string
    */
    public static function singularLabel()
    {
        return __('Page');
    }

    /**
     * Get the fields displayed by the resource.
     *
     * @param    \Illuminate\Http\Request  $request
     * @return  array
     */
    public function fields(Request $request)
    {
        return [
			ID::make( __('ID'),  'id')->rules('required')->sortable(),		
			Text::make( __('Name'),  'name')->rules('required')->sortable(),
			Text::make( __('Identifier'),  'identifier')->rules('required')->hideFromIndex()->sortable()->help(__('cms.help.page.identifier')),
			BelongsTo::make('Metadata')->searchable()->hideFromIndex()->sortable()->nullable(),
			BelongsTo::make('User')->hideFromIndex()->sortable()->default($request->user()->id),
			Translatable::make( __('Slug'),  'slug')->rules('required')->sortable()->help(__('cms.help.general.slug'))->singleLine(),
			Text::make( __('View Name'),  'view_name')->hideFromIndex()->sortable()->help(__('cms.help.page.view_name')),
			Text::make( __('Navigation Type'),  'nav_type')->hideFromIndex()->sortable()->help(__('cms.help.page.nav_type')),
			Boolean::make( __('Published?'),  'published')->rules('required')->sortable()->default(0),
		];
    }

    /**
     * Get the cards available for the request.
     *
     * @param    \Illuminate\Http\Request  $request
     * @return  array
     */
    public function cards(Request $request)
    {
        return [];
    }

    /**
     * Get the filters available for the resource.
     *
     * @param    \Illuminate\Http\Request  $request
     * @return  array
     */
    public function filters(Request $request)
    {
        return [];
    }

    /**
     * Get the lenses available for the resource.
     *
     * @param    \Illuminate\Http\Request  $request
     * @return  array
     */
    public function lenses(Request $request)
    {
        return [];
    }

    /**
     * Get the actions available for the resource.
     *
     * @param    \Illuminate\Http\Request  $request
     * @return  array
     */
    public function actions(Request $request)
    {
        return [];
    }
}

```

Say I create `/resources/lang/en/pages.php` like so:

```
<?php

return [
	'something' => 'something else',
];
```

This causes all Nova routes to throw an exception:

```
htmlspecialchars() expects parameter 1 to be string, array given
```

If I edit the file so it just returns an empty array, the exception disappears.

Oddly enough, the frontend of the site is not affected, and works as expected.

I'm not even sure the nature of the exception itself matters, and may vary by project. It's just that for some reason, Nova is unwantedly trying to read and inject data from this lang file since it has the same name as one of our resources.

Another strange behavior that I noticed prior to the exceptions - but can no longer reproduce due to the exception preventing the view from even rendering - was that instead of outputting the name of the resource in the index listing, it was dumping the raw contents of the `resources/lang/pages.php` into the `<h3>` tag. 

I'm not sure why it went from this behavior to throwing an exception instead. 

---

A separate yet loosely related issue I noticed and just worked around is that Nova can't seem to recognize lang files in `resources/lang/vendor/nova/en/`. 

For example, I created a `help.php` file in that folder, but Nova just kept outputting the translation string key. 

After I moved `help.php` to `resources/lang/en/` and started referencing it in our views as `help.translation_string`, it works fine.

**Comment by zareismail** [comment]
Hi.
This is not about nova. Your translations is wrong. 
You create trans file named 'pages' and calling '__(pages)' translation!
This find pages.php for translation, that is a array. 
You should use '__(pages.pages)' or. Load translation file as json file

**Comment by nathan-io** [comment]
Hi, I think there may be some confusion here.

We aren't calling `__('pages.somekey')` anywhere within Nova.

The entire point of this issue is that Nova was parsing `/resources/lang/pages.php` when there was no call made to it from anywhere within our Nova resources or configuration.

**Comment by nathan-io** [comment]
Ok, so it looks like this is only happening on my local environment, which I discovered tonight after the developer committed a new file `resources/lang/en/metadata.php`. We also have a model and a Nova resource called `Metadata`.

On our staging and production instances (sitting on a Laravel Forge server), there's no issue.

It's only on my local (Ubuntu WSL set up using [valeryan/valet-wsl](https://github.com/valeryan/valet-wsl)) that the new lang file (which, again, we are referencing nowhere within Nova) started throwing the exception.

I'd really like to understand why this is happening only on one environment. 

I even nuked my local instance and re-deployed completely from scratch, just in case. The behavior persists.

**Comment by zareismail** [comment]
> Hi, I think there may be some confusion here.
> 
> We aren't, nor were we ever, calling `__('pages.somekey')` anywhere within Nova.
> 
> The entire point of this issue is that Nova was parsing `/resources/lang/pages.php` when there was no call made to it from anywhere within our Nova resources or configuration.

You called it in the resource label method by `__("Pages")`.
In some system operators, they don't have a difference between uppercase and lowercase strings.
this causes laravel to find `pages.php` file as translation results but you called `Pages`.

**Comment by zareismail** [comment]
> Ok, so it looks like this is only happening on my local environment, which I discovered tonight after the developer committed a new file `resources/lang/en/metadata.php`. We also have a model and a Nova resource called `Metadata`.
> 
> On our staging and production instances (sitting on a Laravel Forge server), there's no issue.
> 
> Yet in my local environment (Ubuntu WSL set up using [valeryan/valet-wsl](https://github.com/valeryan/valet-wsl)) that new lang file (which w'ere referencing only in our frontend views, nowhere within Nova) causes an exception on every Nova route I try to hit. If I rename the file, the issue is resolved.
> 
> I'd really like to understand why this is happening only on one environment.
> 
> I even nuked my local instance and re-deployed completely from scratch, just in case. The behavior persists.

Because of the difference between the system operators. I explained it in the last answer.

**Comment by nathan-io** [comment]
> You called it in the resource label method by __("Pages").

You are correct, this was the issue. I changed this to `return 'Pages'` and `return 'Metadata'` in our respective Nova resources. This resolved the issue.

We've been using [cloudstudio/resource-generator](https://novapackages.com/packages/cloudstudio/resource-generator) to create these resources. I'm not sure if it's standard practice to have `label()` and `singularLabel()` return a translation key instead of a string...

I knew that if you did `__('pages.something')` and the key `something` didn't exist in the `pages.php` lang file for the current locale, Laravel would just output the string "pages".

So I just assumed that if you do `__('pages')`, it will return the string "pages". But apparently it is either returning the entire array, or perhaps just the first key/value pair from it? Either way, that seems a bit counterintuitive.

> Because of the difference between the system operators. I explained it in the last answer.

Yes, this makes sense now. I suppose Ubuntu WSL is ultimately Windows in the end...

Closing the issue. Thanks for your help.
